### PR TITLE
applications: nrf_desktop: Miscellaneous documentation improvements

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces documentation improvements for the NCS 2.3 release.

Jira: NCSDK-19128

### CODE REVIEW

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -1003,8 +1014,11 @@ Default common configuration
 
 The nRF Desktop application aligns the configuration with the nRF Desktop use case by overlaying Kconfig defaults and selecting or implying the required Kconfig options.
 Among others, the Kconfig :ref:`app_event_manager` and :ref:`lib_caf` options are selected to ensure that they are enabled.
+The :ref:`CONFIG_DESKTOP_SETTINGS_LOADER <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_POWER_MANAGER <config_desktop_app_options>` are implied to enable respectively :ref:`nrf_desktop_settings_loader` and :ref:`nrf_desktop_power_manager`.
```
Comment (divipillai) [inline applications/nrf_desktop/README.rst:1017]:
```suggestion
The :ref:`CONFIG_DESKTOP_SETTINGS_LOADER <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_POWER_MANAGER <config_desktop_app_options>` are implied to enable the :ref:`nrf_desktop_settings_loader` and :ref:`nrf_desktop_power_manager` modules, respectively.
```

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -1181,19 +1214,21 @@ To use the nRF Desktop application with your custom board:
...
-       Configure the PWM peripheral in DTS files, and configure the :file:`_def` file of the LEDs module to indicate which PWM channel is assigned to each LED color.
-       Ensure that PWM channels are correctly configured in DTS and PWM driver is enabled in the Kconfig file.
+     * The module uses Zephyr's :ref:`zephyr:led_api` driver for setting the LED color.
+       Zephyr's LED driver can use the implementation based on either GPIO or PWM (Pulse-Width Modulation).
+       The hardware configuration is described through DTS.
+       See the :ref:`caf_leds` configuration section for details.
 
-#. Review Bluetooth options in Kconfig:
+#. Review the :ref:`nrf_desktop_hid_configuration`.
+#. By default nRF Desktop device enables Bluetooth connectivity support.
```
Comment (divipillai) [inline applications/nrf_desktop/README.rst:1223]:
```suggestion
#. By default, the nRF Desktop device enables Bluetooth connectivity support.
```

File: doc/nrf/getting_started/modifying.rst [hunk 3]
```diff
@@ -268,7 +268,10 @@ The Partition Manager's :ref:`static configuration <ug_pm_static>` can also be m
 When the build type has been inferred, the file :file:`pm_static_<buildtype>.yml` will have precedence over :file:`pm_static.yml`.
 
 The child image Kconfig configuration can also be made dependent on the build type.
-The child image Kconfig file is named :file:`<child_image>.conf` instead of :file:`prj.conf`, but otherwise follows the same pattern as the parent Kconfig.
+The child image Kconfig overlay file is named :file:`child_image/<child_image_name>.conf` instead of :file:`prj.conf`, but otherwise follows the same pattern as the parent Kconfig.
+
+Alternatively, the child image Kconfig configuration file can be introduced as :file:`child_image/<child_image_name>/prj.conf` and follow the same pattern as the parent Kconfig.
+For example :file:`child_image/mcuboot/prj_release.conf` can be used to define ``release`` build type for ``mcuboot`` child image.
```
Comment (divipillai) [inline doc/nrf/getting_started/modifying.rst:274]:
```suggestion
For example, :file:`child_image/mcuboot/prj_release.conf` can be used to define ``release`` build type for ``mcuboot`` child image.
```

**Review by divipillai — COMMENTED** [review]
Some minor suggestions.

**Comment by MarekPieta** [comment]
Hi @divipillai
I applied review comments and added a small note related to menuconfig configuration and Kconfig defaults (I think it could help to avoid confusion). Please rereview

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -936,33 +936,51 @@ The nRF Desktop application uses the following files as configuration sources:
...
 Application-specific Kconfig configuration
 ==========================================
 
 The nRF Desktop introduces Kconfig options that can be used to simplify an application configuration.
 These options can be used to select a device role and to automatically apply a default configuration suitable for the selected role.
 
+.. note::
+   Part of the default configuration is applied by modifying the default values of Kconfig options.
+   Keep in mind that changing configuration in menuconfig does not automatically adjust user-configurable values to the new defaults, so that the values must be updated manually.
+   See the section ``Stuck symbols in menuconfig and guiconfig`` on the :ref:`kconfig_tips_and_tricks` in the Zephyr documentation for details.
```
Comment (divipillai) [inline applications/nrf_desktop/README.rst:950]:
```suggestion
   Changing configuration in menuconfig does not automatically adjust user-configurable values to the new defaults. So, you must update those values manually.
   For more information, see the Stuck symbols in menuconfig and guiconfig section on the :ref:`kconfig_tips_and_tricks` in the Zephyr documentation.
```

File: applications/nrf_desktop/README.rst [hunk 5]
```diff
@@ -936,33 +936,51 @@ The nRF Desktop application uses the following files as configuration sources:
...
+
+   The default Kconfig option values are automatically updated if configuration change is applied directly in the configuration files.
```
Comment (divipillai) [inline applications/nrf_desktop/README.rst:952]:
```suggestion
   The default Kconfig option values are automatically updated if configuration changes are applied directly in the configuration files.
```

**Review by divipillai — APPROVED** [review]

File: applications/nrf_desktop/README.rst [hunk 6]
```diff
@@ -1132,38 +1159,52 @@ Adding a new board
 
 When adding a new board for the first time, focus on a single configuration.
 Moreover, keep the default ``debug`` build type that the application is built with, and do not add any additional build type parameters.
+The following procedure uses the gaming mouse configuration as an example.
+
+Zerphyr support for a board
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/README.rst:1164]: Zephyr

**Review by mkapala-nordic — CHANGES_REQUESTED** [review]
Beside one typo - lgtm

**Review by mkapala-nordic — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Pull request merged**